src/main/java/org/openhab/tools/analysis/checkstyle/BuildPropertiesCheck.java

```diff
@@ -0,0 +1,194 @@
+            BIN_INCLUDES_PROPERTY_NAME, BUILD_PROPERTIES_FILE_NAME);
+
+    private static final String MISSING_BIN_INCLUDES_VALUE_MSG = String.format(MISSING_VALUE_MSG,
+            BIN_INCLUDES_PROPERTY_NAME, BUILD_PROPERTIES_FILE_NAME);
+    private static final String MISSING_OUTPUT_VALUE_MSG = String.format(MISSING_VALUE_MSG, OUTPUT_PROPERTY_NAME,
+            BUILD_PROPERTIES_FILE_NAME);
+    private static final String MISSING_SRC_VALUE_MSG = String.format(MISSING_VALUE_MSG, SOURCE_PROPERTY_NAME,
+            BUILD_PROPERTIES_FILE_NAME);
+
+    private static final String EMPTY_FILE_MSG = String.format("Empty %s file", BUILD_PROPERTIES_FILE_NAME);
+
+    // Used for configuration properties
+    private List<String> expectedBinIncludesValues;
+    private List<String> expectedOutputValues;
+    private List<String> possibleSourceValues;
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public BuildPropertiesCheck() {
+        setFileExtensions(PROPERTIES_EXTENSTION);
+    }
+
+    // Configuration property for the expected values for
+    // the bin.includes property in the build.properties file.
+    public void setExpectedBinIncludesValues(String[] binIncludesValues) {
+        this.expectedBinIncludesValues = Arrays.asList(binIncludesValues);
+    }
+
+    // Configuration property for the expected values for
+    // the output property in the build.properties file.
+    public void setExpectedOutputValues(String[] outputValues) {
+        this.expectedOutputValues = Arrays.asList(outputValues);
+    }
+
+    // Configuration property for the possible values for
+    // the source property in the build.properties file.
+    // Some of the possible values will be filtered
+    // if they are not existing in the particular bundle
+    public void setPossibleSourceValues(String[] possibleSourceValues) {
+        // We will have to remove elements from the collection.
+        // Using Arrays.asList returns a list backed by the original which does't allow removal
+        this.possibleSourceValues = new ArrayList<String>(Arrays.asList(possibleSourceValues));
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+        String fileName = file.getName();
+        if (fileName.equals(BUILD_PROPERTIES_FILE_NAME)) {
+            if (!isEmpty(file)) {
+                processBuildProperties(file, lines);
+            } else {
+                log(0, EMPTY_FILE_MSG);
+            }
+        }
+    }
+
+    private void processBuildProperties(File file, List<String> lines) throws CheckstyleException {
+        IBuild buildPropertiesFile = null;
+
+        try {
+            buildPropertiesFile = parseBuildProperties(file);
+        } catch (CheckstyleException e) {
+            logger.error("Exception occured while processiong build.properties file", e);
```

Comment:
Btw does maybe mean that the format of build file is corrupt and should we report that, or do we expect the build to fail in that case.